src/bucket/BucketOutputIterator.cpp

```diff
@@ -136,7 +137,22 @@ BucketOutputIterator::getBucket(BucketManager& bucketManager,
         }
         return std::make_shared<Bucket>();
     }
-    return bucketManager.adoptFileAsBucket(mFilename, mHasher.finish(),
-                                           mObjectsPut, mBytesPut, mergeKey);
+
+    auto hash = mHasher.finish();
+    std::unique_ptr<BucketIndex const> index{};
+
+    // If this bucket needs to be indexed and is not already indexed
+    if (shouldIndex)
+    {
+        if (auto b = bucketManager.getBucketIfExists(hash);
+            !b || !b->isIndexed())
```

Comment:
There actually are a few edge cases where this happens. In particular, this happens when merging a bucket with an empty bucket. This happens whenever the bucketlist trickles down into a previously empty level. Since these are the lowest level buckets, it would be expensive to re-index, so we need to keep this check so we can attach the index to the "new" merge result.

Reply:
makes sense! resolving